src/main/java/timetables_sort_use_case/application_business/AllTimetablesInteractor.java

```diff
@@ -0,0 +1,68 @@
+package timetables_sort_use_case.application_business;
+
+import entities.Timetable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Flow;
+
+/**
+ * A concrete implementation of the AllTimetablesInputBoundary, used in the TimetablesSort use case
+ * subscribers is the list of subscribers for when the user chooses a timetable to further inspect
+ * timetables is the list of timetables that the user is currently presented with, it is updated by
+ * the publisher that this interactor is subscribed to (Timetable Generator Interactor)
+ */
+public class AllTimetablesInteractor implements AllTimetablesInputBoundary,
+        Flow.Publisher<Object>, Flow.Subscriber<Object>{
+    private final ArrayList<Flow.Subscriber<Object>> subscribers = new ArrayList<>();
+    private List<Timetable> timetables = new ArrayList<>();
+
+    public void subscribe(Flow.Subscriber<Object> subscriber) {
+        this.subscribers.add(subscriber);
```

Comment:
There should probably be an @Override here.

Reply:
for some reason intellij screams at me then, saying that it does not override anything
i think it has to do with the type of the subscriber acceptance being object